src/components/Proposal/ActionButton.tsx

```diff
@@ -207,19 +224,19 @@ interface IMyProps {
 
 const ConnectedActionButton = connect(mapStateToProps, mapDispatchToProps)(ActionButton);
 
-export default (props: IMyProps) => {
+export default (props: IMyProps): any => {
 
   const proposalState = props.proposalState;
   let observable: Observable<IRewardState>;
   if (props.currentAccountAddress) {
     observable = proposalState.proposal.rewards({ where: {beneficiary: props.currentAccountAddress}})
       .pipe(map((rewards: Reward[]): Reward => rewards.length === 1 && rewards[0] || null))
-      .pipe(mergeMap(((reward) => reward ? reward.state() : of(null))));
+      .pipe(mergeMap(((reward): Observable<IRewardState> => reward ? reward.state() : of(null))));
```

Comment:
This annotation is wrong, though - it returns `Obserable<IRewardState|null>`. (Again, I vote for turning explicity return type off :-)

Reply:
Thanks for catching that!  I think this is an example of reasons why I strongly support requiring about return-type annotations:  They document the code, which lessens the (re)learning curve for anyone working with the code, they enrichen intellisense feedback which facilitates coding, and they make it clear that `null` is a valid possibility that the code needs to account for, and by such accounting, they reduce risk.

Would be interested in why you would want to ditch this rule.